Source/ARTChannel.m

```diff
@@ -51,6 +51,19 @@ - (void)setOptions:(ARTChannelOptions *)options {
 
 - (void)setOptions_nosync:(ARTChannelOptions *)options {
     _options = options;
+    [self recreateDataEncoderWith:options.cipher];
+}
+
+- (void)recreateDataEncoderWith:(ARTCipherParams*)cipher {
+    dispatch_barrier_async(_queue, ^{
```

Comment:
I believe that `_queue` here ultimately comes from [`-[ARTClientOptions internalDispatchQueue]`](https://github.com/ably/ably-cocoa/blob/902c112d02a34c79d48f00ebf3845ed48d6f08bd/Source/ARTClientOptions.h#L103-L109), which is always a serial queue. That means that `dispatch_barrier_async` behaves the same as `dispatch_async`. I'd suggest that we use the latter to avoid confusion, and for consistency with the rest of the codebase.

Reply:
agree
```objc
_internalDispatchQueue = dispatch_queue_create("io.ably.main", DISPATCH_QUEUE_SERIAL)
```

Reply:
removed here 0b507e371ef454c8b5e6290d1e87afd3db047614